src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -197,6 +196,14 @@ public class OpenIdConnectAuthenticationHandler : AuthenticationHandler<OpenIdCo
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (!string.IsNullOrEmpty(Options.RedirectUri))
```

Comment:
Why not using `message.RedirectUri` instead of `Options.RedirectUri`?
One should be able to replace the `RedirectUri` in the `RedirectToIdentityProvider` notification.